services/authentication/app.py

```diff
@@ -0,0 +1,93 @@
+import os
+
+from flask import Flask, request, jsonify, redirect
+
+from db import Base, engine
+from resources.login import Login
+from resources.roles_updates import RolesUpdates
+from resources.roles import Roles
+
+app = Flask(__name__)
+app.config["DEBUG"] = True
+Base.metadata.create_all(engine)
+
+@app.route('/login/<uuid>/<digid_message>', methods=['GET'])
+def verify_login(uuid, digid_message):
+    try:
+        uuid = int(uuid)
+    except ValueError:
+        return jsonify({'message': 'The id must be an integer'}), 500
+    return Login.digid_check(uuid, digid_message)
+
+@app.route('/login/<uuid>/service_choice', methods=['GET'])
+def provide_service_choices(uuid):
+    try:
+        uuid = int(uuid)
+    except ValueError:
+        return jsonify({'message': 'The id must be an integer'}), 500
+    return Roles.get(uuid)
+
+@app.route('/login/<uuid>/service_choice', methods=['POST'])
+def give_service_choice(uuid):
+    try:
+        uuid = int(uuid)
+    except ValueError:
+        return jsonify({'The id must be an integer'}), 500
+    choice = request.get_json()
+    cookie = Login.create_cookie(uuid, choice)
+
+    if choice['service'] == 'administration':
+        return jsonify({'service': choice['service'], 'cookie': cookie})
+        #return redirect()
+    if choice['service'] == 'party':
+        return jsonify({'service': choice['service'], 'cookie': cookie})
+        #return redirect()
+    if choice['service'] == 'voting':
+        return jsonify({'service': choice['service'], 'cookie': cookie})
+        #return redirect()
+
+@app.route('/user/<uuid>/new_admin_role', methods=['POST'])
+def new_admin_role(uuid):
+    try:
+        uuid = int(uuid)
+    except ValueError:
+        return jsonify({'message': 'The id must be an integer'}), 500
+    return RolesUpdates.add_admin_role(uuid)
+
+@app.route('/user/<uuid>/new_partymember_role', methods=['POST'])
+def new_partymember_role(uuid):
```

Comment:
Do the new_admin_role and new_partymember_role have to be two seperate methods? We could summarize this to 1 accept_new_role call? When you login you should return whether or not there are new roles for the user. If that is the cases the user has the option to accept or decline the role. That is where this call is for. So I feel that the user should give his reply in the body (yes/no) and his answer is communicated to the administration/party management services.